### PR TITLE
Raise hand (un)equipped events on container insert/remove

### LINKED ISSUE #13408: Item actions get duplicated
At least the magboots one is, just equip it and see it show up twice or disappear twice.

### PR DESCRIPTION
**Pull request by Menshin**
## About the PR

Applying an *ElectroJr/Sr* design suggestion : `drop/unequip handshould work more like inventory does, where it just raises unequip events automatically whenever anything is removed from the container` so actions list prediction on client can be accurate.
That fixes sidebar icons duplication, notably for magboots (fixes # 13408).

There's still multiple client reconciliation on a single pickup/drop, but it's now working as intended (as far as i can tell from my local testing).

Alongside the usual maintainers, requesting @ElectroJr review if/when they have some spare time to make sure it was what they had in mind.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs [hunk 1]
```diff
@@ -7,6 +7,25 @@ namespace Content.Shared.Hands.EntitySystems;
...
+        if (!TryGetHand(uid, args.Container.ID, out var hand))
+        {
+            return;
+        }
+
+        var gotUnequipped = new GotUnequippedHandEvent(uid, args.Entity, hand);
+        RaiseLocalEvent(args.Entity, gotUnequipped, false);
+
+        var didUnequip = new DidUnequipHandEvent(uid, args.Entity, hand);
+        RaiseLocalEvent(uid, didUnequip, true);
```
Comment (ElectroJr) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs:26]: I thinks its safe to remove the `broadcast: true` argument, it doesn't seem to be used anywhere. Probably a leftover from some other refactor.

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs [hunk 2]
```diff
@@ -10,6 +10,24 @@ namespace Content.Shared.Hands.EntitySystems;
 
 public abstract partial class SharedHandsSystem : EntitySystem
 {
+    private void InitializePickup()
+    {
+        SubscribeLocalEvent<HandsComponent, EntInsertedIntoContainerMessage>(HandleEntityInserted);
+    }
+    protected virtual void HandleEntityInserted(EntityUid uid, HandsComponent hands, EntInsertedIntoContainerMessage args)
```
Comment (ElectroJr) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs:17]:
```suggestion
    }
    
    protected virtual void HandleEntityInserted(EntityUid uid, HandsComponent hands, EntInsertedIntoContainerMessage args)
```

File: Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs [hunk 3]
```diff
@@ -10,6 +10,24 @@ namespace Content.Shared.Hands.EntitySystems;
...
+        if (!TryGetHand(uid, args.Container.ID, out var hand))
+        {
+            return;
+        }
+
+        var didEquip = new DidEquipHandEvent(uid, args.Entity, hand);
+        RaiseLocalEvent(uid, didEquip, false);
+
+        var gotEquipped = new GotEquippedHandEvent(uid, args.Entity, hand);
+        RaiseLocalEvent(args.Entity, gotEquipped, true);
```
Comment (ElectroJr) [inline Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs:28]: Same for this broadcast

**Review by ElectroJr — CHANGES_REQUESTED** [review]
Haven't tested in game, but other than minor stuff seems fine.

**Comment by Menshin** [comment]
I've been wondering the same for these broadcast, waited for your approval.

Requested changes done.

**Review by ElectroJr — APPROVED** [review]

**Comment by ElectroJr** [comment]
Doesn't seem to fully fix the magboot issue, so I removed the CL
Content Client_qufkBsnBwy

**Comment by Menshin** [comment]
It didn't appeared in local testing, but i guess the misprediction is worse on a live server.

Thanks for removing the CL.

**Pull request merged**